subprojects/soak/src/testFixtures/groovy/org/gradle/launcher/daemon/fixtures/DaemonMultiJdkIntegrationTest.groovy

```diff
@@ -50,18 +50,24 @@ class DaemonMultiJdkIntegrationTest extends DaemonIntegrationSpec {
                 @Override
                 boolean isSatisfiedBy(JvmInstallationMetadata install) {
                     if (version.hasProperty("vendor")) {
-                        if(install.getVendor().getKnownVendor() != version.vendor) {
+                        def actualVendor = install.getVendor().getKnownVendor()
+                        def expectedVendor = version.vendor
+                        if (actualVendor != expectedVendor) {
                             return false
                         }
                     }
-                    return install.languageVersion == version.version
+                    def actualVersion = install.languageVersion.majorVersion
+                    def expectedVersion = version.version
+                    return actualVersion == expectedVersion
                 }
             })
 
             if (jdk != null) {
                 return base
             } else {
-                return EmptyStatement.INSTANCE
```

Comment:
This made the test appear as if it was successful rather than skipped, making it even more confusing.